central/complianceoperator/v2/report/manager/complianceReportgenerator/report_gen.go

```diff
@@ -0,0 +1,38 @@
+package complianceReportgenerator
+
+import (
+	"bytes"
+	"context"
+
+	checkResults "github.com/stackrox/rox/central/complianceoperator/v2/checkresults/datastore"
+	profileDS "github.com/stackrox/rox/central/complianceoperator/v2/profiles/datastore"
+	remediationDS "github.com/stackrox/rox/central/complianceoperator/v2/remediations/datastore"
+	complianceRulesDS "github.com/stackrox/rox/central/complianceoperator/v2/rules/datastore"
+	"github.com/stackrox/rox/generated/storage"
+	"github.com/stackrox/rox/pkg/notifier"
+)
+
+// ReportGenerator interface is used to generate compliance report and send email notification.
+//
+//go:generate mockgen-wrapper
+type ComplianceReportGenerator interface {
+	// ProcessReportRequest will generate a csv report and send notification via email to attached scan config notifiers.
+	ProcessReportRequest(ctx context.Context, req *ComplianceReportRequest) error
+
+	getDataforReport(req *ComplianceReportRequest, ctx context.Context) (*resultEmail, error)
```

Comment:
Why are these interface functions? (I think I left this comment earlier, not sure if it got lost in the shuffle?)